apps/aechannel/src/aesc_state_cache.erl

```diff
@@ -36,6 +57,25 @@ fetch(ChId, PubKey) ->
 delete(ChId) ->
     gen_server:cast(?SERVER, {delete, ChId}).
 
+cache_status(ChId) ->
+    gen_server:call(?SERVER, {cache_status, ChId}).
+
+minimum_depth_achieved(_, ChId, close, _) ->
+    gen_server:call(?SERVER, {min_depth_achieved, ChId, self()}).
+
+table_specs(Mode) ->
+    [
+     {?MODULE, [
+                 aec_db:tab_copies(Mode)
+               , {type, ordered_set}
+               , {record_name, pch}
+               , {attributes, record_info(fields, pch)}
+               , {user_properties, [{vsn, table_vsn(pch)}]}
```

Comment:
I am a bit concerned that this `vsn` property is relied upon as infrastructure in `aec_db` so its definition shall not be here but in `aec_db`.

How does this definition of `vsn` here work with [`aec_db` code that checks `vsn` in existing schema](https://github.com/aeternity/epoch/blob/6bcfe0fa9fac56e928486b4095b94ac7b42d87cb/apps/aecore/src/aec_db.erl#L619-L627)? It looks like this table [gets ignored](https://github.com/aeternity/epoch/blob/6bcfe0fa9fac56e928486b4095b94ac7b42d87cb/apps/aecore/src/aec_db.erl#L602).

Maybe definition of this table shall be in `aec_db` as all other tables - less clean though simpler?

Reply:
There is a hook for checking tables. Adding the requisite code for it.